### PR TITLE
Topics upgrades : add a created_at field and allow to filter on tags or name

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/ecolabdata/ecospheres-front/issues/51

### CODE REVIEW

File: udata/core/topic/models.py [hunk 1]
```diff
@@ -27,6 +28,18 @@ class Topic(db.Document, db.Owned):
     private = db.BooleanField()
     extras = db.ExtrasField()
 
+    created_at = DateTimeField(default=datetime.utcnow, required=True)
```
Comment (maudetes) [inline udata/core/topic/models.py:31]: What about existing ones with `required=True`? Should we fill these somehow?

File: udata/forms/fields.py [hunk 2]
```diff
@@ -540,8 +540,9 @@ def _value(self):
             return ''
 
     def process_formdata(self, valuelist):
-        if not valuelist:
```
Comment (maudetes) [inline udata/forms/fields.py:543]: you may keep this condition. Else a case with valuelist = None won't match the new first condition but then `len(None)` will fail

**Review by maudetes — COMMENTED** [review]
Nice, thank you! Could you add tests on new filters and remove all datasets bugs please? :pray:

File: udata/core/topic/models.py (hunk 1 above)
Comment (quaxsze) [reply udata/core/topic/models.py:31]: `name` is filled by the user and `slug` is filled automaticaly as is `created_at`

Comment (maudetes) [reply udata/core/topic/models.py:31]: I meant existing Topics without a `created_at`

Comment (quaxsze) [reply udata/core/topic/models.py:31]: We can make a ligration but it won't be really relevant. Or remove the `required`

Comment (maudetes) [reply udata/core/topic/models.py:31]: I would say that it is nice to keep the `required`. I've tested your branch and it does fill the created_at value with default value, thus current utcnow.

File: udata/forms/fields.py [hunk 3]
```diff
@@ -541,6 +541,8 @@ def _value(self):
 
     def process_formdata(self, valuelist):
         if not valuelist:
+            if valuelist == []:
```
Comment (maudetes) [inline udata/forms/fields.py:544]: :warning:  valuelist is also `[]` if the field is simply missing! Thus, we're emptying `datasets` if the value is not set. I'm confused about how we're supposed to differentiate between an empty list and a missing value here?

We may de-prioritize this ticket if we go for https://github.com/ecolabdata/ecospheres-front/issues/52.
But I guess this issue is probably not encountered on Topics only?

**Review by maudetes — APPROVED** [review]
:pray:

**Pull request merged**